apisix/core/etcd.lua

```diff
@@ -35,6 +37,55 @@ local is_http = ngx.config.subsystem == "http"
 local _M = {}
 
 
+local admin_api_version
+local function is_v3()
+    if admin_api_version then
+        if admin_api_version == "v3" then
+            return true
+        end
+
+        if admin_api_version == "default" then
+            return false
+        end
+    end
+
+    local local_conf, err = fetch_local_conf()
+    if not local_conf then
+        admin_api_version = "default"
+        log.error("failed to fetch local conf: ", err)
+        return false
+    end
+
+    local api_ver = try_read_attr(local_conf, "apisix", "admin_api_version")
+    if not api_ver or (api_ver and api_ver ~= "v3") then
+        admin_api_version = "default"
+        return false
+    end
+
+    admin_api_version = api_ver
+    return true
+end
+
+
+local function to_v3(body, action)
```

Comment:
`to_v3_action ` ?

Reply:
v3 would remove `action` in response body, `to_v3_action` looks like add `action` in response body.